lib/rubygems/query_utils.rb

```diff
@@ -147,6 +147,11 @@ def show_remote_gems(name)
 
     fetcher = Gem::SpecFetcher.fetcher
 
+    gem_name, gem_versions = from_fetcher(fetcher).slice(:name, :versions).values
+    option_version = options[:version].requirements[0][1]
+
+    raise Gem::MissingSpecError.new(gem_name, option_version) unless gem_versions.include?(Gem::NameTuple.new(gem_name, option_version))
+
```

Comment:
Not sure what the above is trying to do, but what I think we want is:

* Let the index be complete if version is passed, so that we not only get the latest spec, and can filter later the proper version that has been requested. I think you want to add something like this at the beginning of the `execute` method:

```ruby
options[:all] = true if options[:version].specific?
```

The above is just a hacky way to check whether a specific version has been passed, because the default value for version is an open requirement (>= 0).

Then in `show_remote_gems` you probably want to change the filtering to be like this:

```diff
     spec_tuples = if name.nil?
       fetcher.detect(specs_type) { true }
     else
       fetcher.detect(specs_type) do |name_tuple|
-        name === name_tuple.name
+        name === name_tuple.name && options[:version].satisfied_by?(name_tuple.version)
       end
     end
```

Reply:
> Not sure what the above is trying to do, but what I think we want is:

This throws an error if the version passed do not exist.

Reply:
I see. I say let's remove that part for now and focus on getting the two specs you added passing without breaking other things.

Reply:
I've updated files and tests...when you have a chance to look, let me know please.